stream/src/main/java/com/annimon/stream/Stream.java

```diff
@@ -1670,6 +1670,63 @@ public boolean noneMatch(Predicate<? super T> predicate) {
     }
 
     /**
+     * Finds the first element and its index that matches the given predicate.
+     *
+     * <p>This is a short-circuiting terminal operation.
+     *
+     * <p>Example:
+     * <pre>
+     * predicate: (index, value) -&gt; index + value == 7
+     * stream: [1, 2, 3, 4, 5, 2, 0]
+     * index:  [0, 1, 2, 3, 4, 5, 6]
+     * result: Optional.of(IntPair(3, 4))
+     * </pre>
+     *
+     * @param predicate  the predicate to find value
+     * @return an {@code Optional} with {@code IntPair}
+     *         or {@code Optional.empty()} if stream is empty or no value was found.
+     * @since 1.1.8
+     */
+    public Optional<IntPair<T>> findIndexed(IndexedPredicate<? super T> predicate) {
```

Comment:
I have just checked that
```java
<T> void test(Predicate<T> p1, Predicate<T> p2) {
}
```
really doesn't take as parameters `Predicate`s above, and fails with "Incompatible equality constraint: Number and Object". It means that both arguments must be at least of type `T`, and if I make `predicate1` as `Predicate<Number>` it will be still working, but then method will take it when it didn't do it before
So, `Predicate<T>` is equivalent to `Predicate<? extends T>`
With `<? super T>` method works perfectly.